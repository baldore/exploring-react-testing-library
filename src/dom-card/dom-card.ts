type DomCardOptions = {
  title?: string
  content?: string
}

export function init(options: DomCardOptions) {
  const parent = document.createElement('div')

  parent.innerHTML = `
    <h1 data-testid="title">${options.title}</h1>
    <div data-testid="content">${options.content}</div>
  `

  return parent
}
