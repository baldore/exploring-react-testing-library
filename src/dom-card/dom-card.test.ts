import { init } from './dom-card'
import { getByTestId } from '@testing-library/dom'
import faker from 'faker'

test('should have a title', () => {
  const title = faker.commerce.productName()
  const card = init({ title })
  expect(getByTestId(card, 'title')).toHaveTextContent(title)
})

test('should have a content', () => {
  const content = faker.lorem.paragraph()
  const card = init({ content })
  expect(getByTestId(card, 'content')).toHaveTextContent(content)
})

test('should match snapshot', () => {
  const card = init({
    title: 'My Title',
    content: 'My Content',
  })
  expect(card).toMatchSnapshot()
})
