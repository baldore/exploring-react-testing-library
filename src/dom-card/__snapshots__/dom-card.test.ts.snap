// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  
    
  <h1
    data-testid="title"
  >
    My Title
  </h1>
  
    
  <div
    data-testid="content"
  >
    My Content
  </div>
  
  
</div>
`;
